app/models/exercise.rb

```diff
@@ -205,6 +205,10 @@ def study_group_working_time_query(exercise_id, study_group_id, additional_filte
     "
   end
 
+  def teacher_defined_assessment?
+    files.any?(&:teacher_defined_assessment?)
+  end
```

Comment:
I did not know, that there was already a method to check for `teacher_defined_tests`. 
So thanks @MrSerth for the much cleaner solution. :)